js/web/plunderer/js/plunderer.js

```diff
@@ -482,14 +533,17 @@ let Plunderer = {
 
 		let todaySP = 0;
 		let thisWeekSP = 0;
-		let totalSPSelect = filterByPlayerId ?
-			(Plunderer.db.actions.where('playerId').equals(filterByPlayerId)) :
-			(Plunderer.db.actions.where('type').equals(Plunderer.ACTION_TYPE_PLUNDERED));
+		let totalSPSelect = await (filterByPlayerId ?
+			Plunderer.db.actions.where('playerId').equals(filterByPlayerId) :
+			Plunderer.db.actions.where('type').equals(Plunderer.ACTION_TYPE_PLUNDERED)).toArray();
 
 		let totalSP = 0;
 
-		await totalSPSelect.each((it) => {
-			const sp = (it.sp || 0);
+		for(const it of totalSPSelect)
+		{
+			const factor = await Plunderer.getActionFactor(it);
```

Comment:
Hi, I think this should be optimized. Try to avoid async operations in loop.

My suggestion create Key-Value lookup table

```js
const factorKV = (await Plunderer.db.plunderAndPillages.toArray()).reduce((acc, val) => {
  acc[val.plunderId] = val;
  return acc;
}, {});

const getFactor(plunderId) => factorKV[plunderId] ? 2 : 1;

// .... 
for(const it of totalSPSelect) {
  const sp = (it.sp || 0) * getFactor(it.plunderId);
```